### ISSUE TITLE
update_file_custom_metadata(..., is_metadata_file=False): UnboundLocalError: local variable 'is_metadata' referenced before assignment

### ISSUE THREAD

**Issue by 2-5** [dask/fastparquet#829]
**Describe the issue**:

`writer.update_file_custom_metadata` fails if it's `is_metadata_file` argument is used.

**Minimal Complete Verifiable Example**:

```python
import pandas as pd

import fastparquet
from fastparquet import writer

df = pd.DataFrame([1, 2, 3], columns=["a"])
fastparquet.write("data.parquet", df)

metadata = {"version": "1"}

writer.update_file_custom_metadata("data.parquet", custom_metadata=metadata, is_metadata_file=False)
# Traceback (most recent call last):
#   File "test.py", line 11, in <module>
#     writer.update_file_custom_metadata("data.parquet", custom_metadata=metadata, is_metadata_file=False)
#   File ".pyenv\lib\site-packages\fastparquet\writer.py", line 1545, in update_file_custom_metadata
#     if is_metadata:
# UnboundLocalError: local variable 'is_metadata' referenced before assignment

writer.update_file_custom_metadata("data.parquet", custom_metadata=metadata)
# WORKS
```

**Bug location**

https://github.com/dask/fastparquet/blob/main/fastparquet/writer.py#L1574-L1580

**Environment**:

- fastparquet: 2022.11.0
- Python version: 3.10
- Operating System: Windows
- Install method (conda, pip, source): pip

**Comment by yohplala** [comment]
Take.

**Comment by yohplala** [comment]
Fixed by #830
